person.rb

```diff
@@ -4,16 +4,16 @@
 require_relative 'trimmer_decorator'
 require_relative 'rental'
 class Person < Nameable
-  attr_accessor :name, :age
-  attr_reader :id, :rentals
+  attr_accessor :name, :age, :rentals
+  attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
-    @id = SecureRandom.uuid
+    @id = generate_uuid_as_hexadecimal
```

Comment:
[OPTIONAL]
- [ ] Right now your generated id seems to be a very long figure. I'd generally advice you make use of ruby's inbuilt random number generator. see the code snippet below
- code snippet
```ruby
@id = Random.rand(1...1000) 
```